components/Flashcard/Flashcard.js

```diff
@@ -1,19 +1,43 @@
 import styled from "styled-components";
 import { useState } from "react";
 
-export default function Flashcard({ question, answer, collection }) {
+export default function Flashcard({ flashcard, onIsCorrect }) {
   const [showAnswer, setShowAnswer] = useState(false);
+  const [isCorrect, setIsCorrect] = useState(flashcard.isCorrect);
+
+  const { question, answer, collectionTitle: collection, id } = flashcard;
 
   function handleShowAnswer() {
     setShowAnswer(!showAnswer);
   }
 
+  function handleToggleIsCorrect() {
+    setIsCorrect(!isCorrect);
+    onIsCorrect(id);
+  }
+
   return (
     <CardContainer onClick={handleShowAnswer}>
       <StyledFlashcard $showAnswer={showAnswer}>
         <CardFront>
           <CollectionTitle>{collection}</CollectionTitle>
           <Question>{question}</Question>
+          {isCorrect && (
+            <StyledButton
+              $isCorrect={isCorrect}
+              onClick={handleToggleIsCorrect}
+            >
+              <svg
+                xmlns="http://www.w3.org/2000/svg"
+                height="24px"
+                viewBox="0 -960 960 960"
+                width="24px"
+                fill="#000"
+              >
+                <path d="m256-200-56-56 224-224-224-224 56-56 224 224 224-224 56 56-224 224 224 224-56 56-224-224-224 224Z" />
+              </svg>
```

Comment:
you can move this svg into its own file, e.g. "IsCorrectIcon.svg" and import it like a component, this would clean up the code a bit:
```js
// IsCorrectIcon.svg
 <svg
                xmlns="http://www.w3.org/2000/svg"
                height="24px"
                viewBox="0 -960 960 960"
                width="24px"
                fill="#000"
              >
                <path d="m256-200-56-56 224-224-224-224 56-56 224 224 224-224 56 56-224 224 224 224-56 56-224-224-224 224Z" />
              </svg>
```

```js
import Icon from './IsCorrectIcon.svg'

...

<Icon />
```

Reply:
Thanks for the input, we adapted this change.